### ISSUE TITLE
assertEquals doesn't show diff for strings

### ISSUE THREAD

**Issue by mekras** [sebastianbergmann/phpunit#1271]
PHPUnit 4.1.1, Comparator 1.0.0.

assertEquals('foo', 'bar') outputs only

```
PHPUnit_Framework_ExpectationFailedException : Failed asserting that two strings are equal.
```

And there is no info about **what** strings are not equal.

It seems that [here](https://github.com/sebastianbergmann/phpunit/blob/master/src/Framework/Constraint/IsEqual.php#L172) instead of

``` php
trim($description . "\n" . $f->getMessage()),
```

should be

``` php
trim($description . "\n" . $f->toString()),
```

And in this case output will be:

```
PHPUnit_Framework_ExpectationFailedException : Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'foo'
+'bar'
```

**Comment by dossy** [comment]
I know things have been this way for a long time, now, but it would still be useful to see what the expected and actual values were, otherwise you end up resorting to having to temporarily instrument your tests with `var_dump()`s to see what the actual value was, which is tedious and annoying.

**Comment by dossy** [comment]
FWIW, the latest documentation for PHPUnit still suggests that seeing the actual and expected values in the test runner output is the expected behavior:

https://phpunit.de/manual/current/en/appendixes.assertions.html#appendixes.assertions.assertEquals

**Comment by dossy** [comment]
Okay, for the benefit of anyone else who stumbles onto this after me, I just discovered that these two things do NOT result in the same outcome:

A:

```php
try {
    $actual = somePotentiallyExceptionThrowingFunc();
    $this->assertEquals($expected, $actual);
} catch (Exception $e) {
    $this->fail($e->getMessage());
}
```

And, B:

```php
$actual = null;

try {
    $actual = somePotentiallyExceptionThrowingFunc();
} catch (Exception $e) {
    $this->fail($e->getMessage());
}

$this->assertEquals($expected, $actual);
```

When the assertion is inside the `try` block, you only get the one line, `Failed asserting that two strings are equal.`  When the assertion is outside the `try` block, you get that line, plus the expected/actual diff.

I suppose the `catch` is over-broad and eating the `PHPUnit_Framework_ExpectationFailedException` that the assertion is throwing?  I can catch that in my tests and re-throw it, but this certainly caught me by surprise.

Hopefully this helps someone else who made the same mistake I did.
